admin_manual/installation/nginx.rst

```diff
@@ -114,75 +76,109 @@ webroot of your nginx installation. In this example it is
+      # which would otherwise handle requests for `/.well-known`.
+      location ^~ /.well-known {
+          # The following 6 rules are borrowed from `.htaccess`
+      
+          rewrite ^/\.well-known/host-meta\.json  /public.php?service=host-meta-json  last;
+          rewrite ^/\.well-known/host-meta        /public.php?service=host-meta       last;
+          rewrite ^/\.well-known/webfinger        /public.php?service=webfinger       last;
+          rewrite ^/\.well-known/nodeinfo         /public.php?service=nodeinfo        last;
+          
+          location /.well-known/carddav   { return 301 /remote.php/dav/; }
+          location /.well-known/caldav    { return 301 /remote.php/dav/; }
       }
-
-      location ~ ^\/(?:index|remote|public|cron|core\/ajax\/update|status|ocs\/v[12]|updater\/.+|oc[ms]-provider\/.+|.+\/richdocumentscode\/proxy)\.php(?:$|\/) {
-          fastcgi_split_path_info ^(.+?\.php)(\/.*|)$;
+      
+      # Rules borrowed from `.htaccess` to hide certain paths from clients
+      location ~ ^/(build|tests|config|lib|3rdparty|templates|data)($|/)  { return 404; }
+      location ~ ^/(autotest|occ|issue|indie|db_|console)                 { return 404; }
+      
+      # Respond with HTTP 404 to *all* requests for hidden files;
+      # not just hidden files that reside in the root directory.
+      location ~ /\. { return 404; }
+      
+      # Ensure this block, which passes PHP files to the PHP process, is above the blocks
+      # which handle static assets (as seen below). If this block is not declared first,
+      # then Nginx will encounter an infinite rewriting loop when it prepends `/index.php`
+      # to the URI, resulting in a HTTP 500 error response.
+      location ~ \.php($|/) {
+          fastcgi_split_path_info ^(.+?\.php)(/.*|)$;
           set $path_info $fastcgi_path_info;
+          
           try_files $fastcgi_script_name =404;
+          
           include fastcgi_params;
           fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
           fastcgi_param PATH_INFO $path_info;
           fastcgi_param HTTPS on;
-          # Avoid sending the security headers twice
-          fastcgi_param modHeadersAvailable true;
-          # Enable pretty urls
-          fastcgi_param front_controller_active true;
+          
+          fastcgi_param modHeadersAvailable true;         # Avoid sending the security headers twice
+          fastcgi_param front_controller_active true;     # Enable pretty urls
           fastcgi_pass php-handler;
+          
           fastcgi_intercept_errors on;
           fastcgi_request_buffering off;
       }
-
-      location ~ ^\/(?:updater|oc[ms]-provider)(?:$|\/) {
-          try_files $uri/ =404;
-          index index.php;
+      
+      location ~ \.(css|js|svg|gif)$ {
+          try_files $uri $uri/ /index.php$request_uri;
+          expires 6M;         # Cache-Control policy borrowed from `.htaccess`
```

Comment:
Within this `location` block, we use `expires` rather than `add_header Cache-Control`, so to my knowledge this does not override any/all `add_header` directives in the outer `server` scope — I'll check this, though

Reply:
Tested this now, and it does behave as I expected; headers are served correctly, e.g. for `/apps/files_pdfviewer/css/style.css` used on the login page, I get:

```
cache-control: max-age=15552000
date: Sun, 12 Jul 2020 19:59:38 GMT
expires: Fri, 08 Jan 2021 19:59:38 GMT
last-modified: Tue, 26 May 2020 08:09:01 GMT
referrer-policy: no-referrer
x-content-type-options: nosniff
x-download-options: noopen
x-frame-options: SAMEORIGIN
x-permitted-cross-domain-policies: none
x-robots-tag: none
x-xss-protection: 1; mode=block
```

Of course, for requests that are handled by the front-end controller, such as `/css/core/xxxx-xxxx-server.css` used after logging in, the default expiration policy of 1 minute specified in the `server` block applies, instead of 6 months.

Reply:
Sounds good, I wonder why we had to make this so complicated before then, lol. I didn't do it....
Again, if you could give me a few days to test your config, I would appreciate that.

Reply:
Confirmed what you found. `expires` works as expected, unlike `add header`. I'll go ahead and remove the note that is now irrelevant.